tests/gateways/test_repodata_gateway.py

```diff
@@ -128,6 +129,37 @@ def test_stale(tmp_path):
     assert not cache.state.mod
     assert not cache.state.etag
 
```

Comment:
@dholth we can split this test into 2 tests.. 

1. matching `mod` 
2.  `wrongly-typed mod to empty string`